providers/cloudflare/connection/connection.go

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+package connection
+
+import (
+	"errors"
+	"os"
+
+	"github.com/cloudflare/cloudflare-go"
+	"go.mondoo.com/cnquery/v11/providers-sdk/v1/inventory"
+	"go.mondoo.com/cnquery/v11/providers-sdk/v1/plugin"
+)
+
+const (
+	OPTION_API_TOKEN = "api-token"
+)
+
+type CloudflareConnection struct {
+	plugin.Connection
+	Conf  *inventory.Config
+	asset *inventory.Asset
+
+	Cf *cloudflare.API
+}
+
+func NewCloudflareConnection(id uint32, asset *inventory.Asset, conf *inventory.Config) (*CloudflareConnection, error) {
+	conn := &CloudflareConnection{
+		Connection: plugin.NewConnection(id, asset),
+		Conf:       conf,
+		asset:      asset,
+	}
+
+	// initialize your connection here
+	token := conf.Options[OPTION_API_TOKEN]
+	if token == "" {
+		token = os.Getenv("CLOUDFLARE_TOKEN")
+		if token == "" {
+			return nil, errors.New("API Token is required")
```

Comment:
you should make the error more actionable. For example, https://github.com/mondoohq/cnquery/blob/main/providers/github/provider/provider.go#L75